test/controllers/songs_controller_test.rb

```diff
@@ -35,11 +37,40 @@ class SongsControllerTest < ActionController::TestCase
     assert_includes(songs_data, songs(:glorious_day))
   end
 
-  test "should get new" do
+  # "new" action tests
+  test "new song page should load successfully" do
     get :new
     assert_response :success
   end
 
+  # "create" action tests
+  test "should load the new song template when song creation unsuccessful" do
+    post :create, song: {problem: true}
+    assert_template :new
+  end
+
+  test "should notify user appropriately when song creation unsuccessful" do
+    post :create, song: {problem: true}
+    assert_not_nil flash[:error]
+  end
+
+  test "submitting a valid song should result in a new song in the database" do
```

Comment:
Did you try putting a `byebug` statement in the create action to see if the song was actually saving? Or maybe  change @song.save to @song.save! for debugging so that an exception will be raised if there are any errors on save.